sound/soc/intel/boards/skl_hda_dsp_generic.c

```diff
@@ -168,6 +168,26 @@ static int skl_hda_fill_card_info(struct snd_soc_acpi_mach_params *mach_params)
 	return 0;
 }
 
+#define HDA_ANA_CODEC_AUTOSUSPEND_DELAY_MS 2000
+
+static void skl_hda_set_ana_codec_autosuspend(struct snd_soc_card *card)
+{
+	struct snd_soc_dai *codec_dai;
+	struct hdac_hda_priv *hda_pvt;
+	struct hdac_device *hdev;
+
+	codec_dai = snd_soc_card_get_codec_dai(card, "Analog Codec DAI");
+	if (!codec_dai)
+		return;
+
+	hda_pvt = snd_soc_component_get_drvdata(codec_dai->component);
+	hdev = &hda_pvt->codec.core;
+
+	pm_runtime_set_autosuspend_delay(&hdev->dev,
+					 HDA_ANA_CODEC_AUTOSUSPEND_DELAY_MS);
+	pm_runtime_use_autosuspend(&hdev->dev);
```

Comment:
@jason77-wang do we need a pm_runtime_mark_last_busy() after setting the autosuspend delay? Is the codec runtime active at this point or it is suspended?